macros/cross_db_utils/width_bucket.sql

```diff
@@ -0,0 +1,59 @@
+{% macro width_bucket(expr, min_value, max_value, num_buckets) %}
+  {{ adapter_macro('dbt_utils.width_bucket', expr, min_value, max_value, num_buckets) }}
+{% endmacro %}
+
+
+{% macro default__width_bucket(expr, min_value, max_value, num_buckets) -%}
+
+    {% set bin_size -%}
+    (( {{ max_value }} - {{ min_value }} ) / {{ num_buckets }} )
+    {%- endset %}
+    (
+        -- to break ties when the amount is eaxtly at the bucket egde
+        case
+            when 
+                mod(
+                    {{ dbt_utils.safe_cast(expr, 'decimal(28,6)') }},
+                    {{ dbt_utils.safe_cast(bin_size, 'decimal(28,6)') }}
+                ) = 0
+            then 1
+            else 0
+        end
+    ) +
+      -- Anything over max_value goes the N+1 bucket
+    least(
+        ceil(
+            ({{ expr }} - {{ min_value }})/{{ bin_size }}
+        ),
+        {{ num_buckets }} + 1
+    )
+{%- endmacro %}
+
+{% macro redshift__width_bucket(expr, min_value, max_value, num_buckets) -%}
+
+    {% set bin_size -%}
+    (( {{ max_value }} - {{ min_value }} ) / {{ num_buckets }} )
+    {%- endset %}
+    (
+        -- to break ties when the amount is eaxtly at the bucket egde
+        case
+            when 
```

Comment:
AFAIK, `mod` works on Redshift. Is there a reason you aren't using it here?

Reply:
Oh I got it -- because `mod` only works on two integers. Yeesh.
```
Function "mod(numeric,numeric)" not supported.
```
Even though the [docs](https://docs.aws.amazon.com/redshift/latest/dg/r_MOD.html) indicate it should work:
> The first input parameter is an INTEGER, SMALLINT, BIGINT, or DECIMAL number. If either parameter is a DECIMAL type, the other parameter must also be a DECIMAL type.

Yikes!